bindgen/clang.rs

```diff
@@ -1524,6 +1564,42 @@ impl SourceLocation {
     }
 }
 
+impl SourceLocation {
+    /// Compare source locations, also considering `#include` directives.
+    pub(crate) fn partial_cmp_with_context(
```

Comment:
Could you expand the docs for this function stating how the comparison works? e.g: built-in locations always come first, two locations in the same file are compared by the offset, two locations in different files are compared by inclusion.